apis/python/src/tiledbsoma/tiledb_object.py

```diff
@@ -113,15 +125,100 @@ def exists(self) -> bool:
         # before a third, successful HTTP request for group-open.  Instead, we directly attempt the
         # group-open request, checking for an exception.
         try:
-            return (
-                self._metadata.get(util.SOMA_OBJECT_TYPE_METADATA_KEY) == self.soma_type
-            )
+            with self._maybe_open():
+                return (
+                    self._metadata.get(util.SOMA_OBJECT_TYPE_METADATA_KEY)
+                    == self.soma_type
+                )
         except tiledb.cc.TileDBError:
             return False
 
-    @abstractmethod
-    def _tiledb_open(self, mode: str = "r") -> Union[tiledb.Array, tiledb.Group]:
-        """Open the underlying TileDB array or Group"""
+    def open(self: TTileDBObject, mode: str = "r") -> TTileDBObject:
```

Comment:
Should abstract variants of these methods/types be pushed upstream to `somacore`?  Or is that a later PR?

Reply:
Can I defer to @thetorpedodog to look at how much of this should lift up into `somacore` since he plans some ~immediate refactoring anyway?

(Possibly most of the tiledb_object.py diff should lift up, since it's just managing the state transitions and exception handling in a totally generic way.)

Reply:
Yes, this seems sensible, and in another PR @thetorpedodog mentioned this was his plan.  Resolving this comment.